### DISCUSSION TITLE
Bug: Text Generation Process Fails Silently in Ollama Serve

### DISCUSSION THREAD

**Discussion by hemangjoshi37a** [open-webui/open-webui#514] [General]
**Issue Description:**
At times, when I use the "generate" button to initiate text generation after entering a prompt, the process fails to produce any output. This failure occurs without any error messages. Observing the `htop` terminal, it appears that the `ollama serve` process either stops or completes. However, the chat window continues to display the shimmer effect, which usually indicates ongoing generation, leading to confusion about the actual state of the process.

**Steps to Reproduce:**
1. Enter a text prompt in the chat window.
2. Click on the "generate" button to start the text generation process.
3. Observe the lack of output in the chat window and the status of the process in the `htop` terminal.

**Expected Behavior:**
When the text generation process is initiated, it should either successfully generate text or provide an error message if it fails.

**Actual Behavior:**
The text generation process stops or completes silently without any output in the chat window, while the `ollama serve` process appears to stop in the `htop` terminal. The chat window continues to show the shimmer effect, incorrectly suggesting that generation is still in progress.

**Environment:**
- python 3.11 and latest ollama version installed yesterday using github readme.

**Comment by tjbck** [comment]
Cannot reproduce the issue, moving to discussion. Please provide us with the step-by-step guide on how to reproduce the issue, thanks!

**Reply by dfv219** [reply]
i did daily updates with git pull and also did update the install of the requirements, since I replied here, but the issue is still existing. Ui is not usable anymore. What kind of information would help you to reproduce the issue?
I am using Ubuntu 22.04 LTS on WSL2 on Windows.

**Reply by tjbck** [reply]
@dfv219 are you also rebuilding the frontend as well? browser console logs will be extremely helpful for us!

**Reply by dfv219** [reply]
@tjbck I'm not a dev, i am just a curious User and I wanted to run ollama and ollama webui at home. Should i post the logs just here in plaintext as an answer? It seems to me that those console logs are very long texts. Sorry but I've never done that before, I don't know it better. (The GET https://127.0.0.1:8080/openai..... i changed the external api to the internal link just not to send many bad requests to openai servers) (And the last working chat I have in my history has the timestamp 1705652064 )

but just to make sure i understood what you are talking about:
0.237376f9.js:1 {status: true, version: 'v1.0.0-alpha.61', auth: true, default_models: null, default_prompt_suggestions: Array(4)}
2.d1d57bd5.js:52 IDB Not Found
2.d1d57bd5.js:52 0.1.20
2.d1d57bd5.js:35 settings true
3.7b071185.js:1 initNewChat
3.7b071185.js:1 
3.7b071185.js:1 {status: true, version: 'v1.0.0-alpha.61', auth: true, default_models: null, default_prompt_suggestions: Array(4)}
start.b54b2260.js:1 
        
        
        GET http://127.0.0.1:8080/openai/api/models 401 (Unauthorized)
window.fetch @ start.b54b2260.js:1
(anonymous) @ inspector.js:7
window.fetch @ inspector.js:7
d @ index.fe45cfdc.js:1
_ @ 2.d1d57bd5.js:52
await in _ (asynchron)
(anonymous) @ 2.d1d57bd5.js:52
b @ index.d1cf4fe0.js:1
(anonymous) @ 2.d1d57bd5.js:52
await in (anonymous) (asynchron)
z @ scheduler.4bc66f67.js:1
(anonymous) @ index.369d03c1.js:4
lt @ scheduler.4bc66f67.js:1
Promise.then (asynchron)
it @ scheduler.4bc66f67.js:1
ht @ index.369d03c1.js:4
(anonymous) @ index.369d03c1.js:4
(anonymous) @ 0.237376f9.js:1
await in (anonymous) (asynchron)
z @ scheduler.4bc66f67.js:1
(anonymous) @ index.369d03c1.js:4
lt @ scheduler.4bc66f67.js:1
Mt @ index.369d03c1.js:4
ue @ app.efd3f808.js:1
Pe @ start.b54b2260.js:1
ce @ start.b54b2260.js:1
await in ce (asynchron)
re @ start.b54b2260.js:1
goto @ start.b54b2260.js:1
Nt @ start.b54b2260.js:3
(anonymous) @ (Index):47
Promise.then (asynchron)
(anonymous) @ (Index):46
index.fe45cfdc.js:1 {detail: "Oops! It looks like there's a hiccup. The API key …o provide a valid API key to access this feature."}
2.d1d57bd5.js:52 OpenAI: Network Problem
2.d1d57bd5.js:35 {models: Array(1)}
3.7b071185.js:1 submitPrompt 
3.7b071185.js:1 []
3.7b071185.js:2 mistral:latest
3.7b071185.js:4 {"id": "1de8f64b-e97e-487e-8953-8476fa605d5f", "done": false}
3.7b071185.js:5 TypeError: Cannot read properties of undefined (reading 'content')
    at _e (3.7b071185.js:4:134)
    at async 3.7b071185.js:2:240
    at async Promise.all (:8080/index 0)
    at async W (3.7b071185.js:2:113)
    at async we (3.7b071185.js:1:7039)

**Comment by dfv219** [comment]
I have the same problem. since i updated to the last version like 3 days ago, i got the message "Bad substitution" when starting with "sh start.sh" after using bash start.sh the error was gone, but the behaviour of the ui is exactly the same as hemangjoshi37a wrote. Sorry i am not a dev and i can't provide any special information unless someone tells me exactly what he needs to know.
![20240123_140746-Ollama-webui-behaviour](https://github.com/ollama-webui/ollama-webui/assets/13416470/2f4cf6f3-a782-4130-9b3e-610816b2d313)

**Comment by llagerlof** [comment]
This is happening to me, but it's not a open-webui problem, since it's happening also on command line using ollama. Sometimes nothing is returned.

**Reply by llagerlof** [reply]
My fix was upgrade ollama to the latest version.

**Comment by hemangjoshi37a** [comment]
I simply solved by upgrading to latest version of `open-webui`
